source/MissionAction.cpp

```diff
@@ -392,3 +397,39 @@ int64_t MissionAction::Payment() const noexcept
 {
 	return action.Payment();
 }
+
+
+
+shared_ptr<ConditionSet> MissionAction::GetToDisplay(const DataNode &node)
+{
+	shared_ptr<ConditionSet> conditions;
+	for(auto &child : node)
+		if(child.Size() != 2 || child.Token(0) != "to" || child.Token(1) != "display" || !child.HasChildren())
+			child.PrintTrace("Ignoring unrecognized dialog token");
+		else if(!conditions)
+			conditions = make_shared<ConditionSet>(child);
+		else
+			conditions->Load(child);
+	return conditions;
+}
+
+
+
+void MissionAction::LoadDialog(const DataNode &node)
+{
+	// Parse the "dialog phrase whatever" and "dialog whatever" lines:
+	if(node.Size() == 3 && node.Token(1) == "phrase")
+		dialog.emplace_back("${" + node.Token(2) + "}", nullptr);
+	else if(node.Size() == 2)
+		dialog.emplace_back(node.Token(1), nullptr);
+	// Parse dialog in children:
+	for(auto &child : node)
+		if(child.Size() == 2 && child.Token(0) == "phrase")
+			dialog.emplace_back("${" + child.Token(1) + "}", GetToDisplay(child));
+		else
+		{
+			if(child.Size() > 1)
+				child.PrintTrace("Ignoring extra tokens.");
+			dialog.emplace_back(child.Token(0), GetToDisplay(child));
+		}
+}
```

Comment:
This doesn't appear to support an anonymous phrase defined as a child of the dialog node.
The storage for such an anonymous phrase (`ExclusiveItem<Phrase> dialogPhrase`) has also been removed from the header.

Reply:
That has been corrected.